### ISSUE TITLE
[Android] Handle QR code for sync when Google Play Services aren't available

### ISSUE THREAD

**Issue by jsecretan** [brave/brave-browser#11477]
## Description 
Android's built-in QR code scanning depends on Google Play services. Right now, attempting to use the QR code scanning feature brings up an error message about Brave not running without Google Play services installed. Brave is a privacy browser, so it would be nice to have better handling for for devices that do not have Google Play services available.

One option is to hide the QR code scanning altogether. This is not ideal but better than throwing an error to the user.

Another, more preferable option is to delegate scanning to the `com.google.zxing.client.android.SCAN` intent. It appears to be widely used as per [this tutorial](https://tekeye.uk/android/examples/scan-barcode-from-android-app) and [examples in the wild](https://github.com/wewewe718/QrAndBarcodeScanner/blob/master/app/src/main/AndroidManifest.xml#L71).

**Comment by AlexeyBarabash** [comment]
+1 I can reproduce

**Comment by AlexeyBarabash** [comment]
![Screenshot from 2020-08-27 18-55-51](https://user-images.githubusercontent.com/24739341/91469181-bdcb1780-e89b-11ea-88ca-d8ae970fc971.png)

**Comment by AlexeyBarabash** [comment]
+1 from https://twitter.com/Don_Timeless/status/1358648921522917376

**Comment by antonok-edm** [comment]
Related: https://github.com/brave/brave-browser/issues/12067

**Comment by AlexeyBarabash** [comment]
+1 from 
https://community.brave.app/t/degoogled-android-device-asking-for-play-services/651069

**Comment by Brave-Matt** [comment]
Yeah @AlexeyBarabash @timchilds think we can get this sorted out? Even if we don't auto-detect that GPlay services aren't enabled can we at least get it so that the error message can be dismissed and allow the user to select the code words option?

**Comment by SomeAsian6uy** [comment]
If you spam "Enter code words" it dismisses the device not compatible box and brings you to the code entry menu. I was able to enter the code by spamming the keyboard to enter the code in between the not supported dialogue box and the eventual app not responding box. It does sync tho.

Edit: Stopped syncing eventually. Not sure if its a different issue. It shows the other sync devices but the other synced devices does not show the device without google services.

**Comment by Nikvothe** [comment]
This is still a problem? This is ridiculous guys come on, Brave is supposed to be on the good side isn't that right?
Especially now that Google is starting to lock any download alternatives other than Google Play, I would wager more people are moving towards open roms like LineageOS or GrapheneOS.

I fully agree with the proposal given by @Brave-Matt

**Comment by dark-penguin** [comment]
Could I suggest at least not retrying the action as a stopgap measure, just to make it work for now? I've seen other apps give the same message; once the message is dismissed, it doesn't show up again.

If that the message box shows up just once, then at least you can enter the words after you dismiss it. Right now, it keeps spamming _even on the "words" screen_, so setting up sync is completely impossible. This would at least make Brave _compatible with non-proprietary operating systems_ .

**Comment by dark-penguin** [comment]
Here, fixed it myself. Actually adding a dependency on a libre QR code scanning library would probably never be implemented, but this way at least setting up sync is possible - we can handle the rest.

**Comment by greglechin** [comment]
Why is there no response from the devs on this?

**Comment by ypurpl** [comment]
while code entry does work, the device not compatible dialog is constantly spammed and makes it nearly impossible to enter a code.
managed to do it by rapidly clicking

**Comment by srirambv** [comment]
Verification passed on OnePlus 6T with Linage OS 22.2 running

Brave | 1.94.114 Chromium: 152.0.7977.54 (Official Build) (64-bit)
-- | --
Revision | c58d45f31c7ebd409699a6b602191e553e1cfb4a
OS | Android 15; Build/BP1A.250505.005; 35; REL

- Verified steps from [brave-core#38292](https://github.com/brave/brave-core/pull/38292)
- Verified trying to scan a Sync QR code shows the error dialog 
- Verified dismissing the error dialog doesn't bring it back up
- Verified rotating the device several times including a 180 deg rotation doesn't bring the error message 
- Verified entering the sync code words works as expected and joins the Sync chain
- Verified no crash when the error dialog is shown 
- Verified unable to scan QR code in URL bar as well

https://github.com/user-attachments/assets/2ef3bcd1-aeec-40e9-b949-844b139786f6

Verification passed on Google Pixel 8 running 

Brave | 1.94.115 Chromium: 152.0.7977.54 (Official Build) (64-bit) 
-- | --
Revision | 65849aa13268dbd77e64c5638a8d378f893c14da
OS | Android 17; Build/CP2A.260805.005; 37; REL

- Verified able to scan QR code and join Sync chain
- Verified rotating the device on Sync scan screen retains the preview upright
- Verified when scanning switching to other apps and back retains the preview
- Verified scanning QR code in URL bar loads the URL 
- Verified revoking camera permission and opening the Sync scanner brings the permission prompt back
- Verified granting the permission to use Camera loads the preview and scans the QR code correctly 

https://github.com/user-attachments/assets/fe3a3e77-8718-46d9-8937-01e54cf61680
